core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -984,7 +1051,8 @@ object Cause extends Serializable {
         else loop(stack.head, stack.tail, parallel, sequential)
     }
 
-    loop(c, List.empty, Set.empty, List.empty)
+    if (c eq Empty) (Set.empty, Nil)
```

Comment:
We could probably avoid the tuple allocation here but this method isn't used much